x/wasm/internal/keeper/handler_plugin_encoders.go

```diff
@@ -225,32 +225,33 @@ func EncodeWasmMsg(sender sdk.AccAddress, msg *wasmvmtypes.WasmMsg) ([]sdk.Msg,
 	}
 }
 
-func EncodeIBCMsg(ctx sdk.Context, sender sdk.AccAddress, contractIBCPortID string, msg *wasmvmtypes.IBCMsg) ([]sdk.Msg, error) {
-	switch {
-	case msg.CloseChannel != nil:
-		return []sdk.Msg{&channeltypes.MsgChannelCloseInit{
-			PortId:    PortIDForContract(sender),
-			ChannelId: msg.CloseChannel.ChannelID,
-			Signer:    sender.String(),
-		}}, nil
-	case msg.Transfer != nil:
-		amount, err := convertWasmCoinToSdkCoin(msg.Transfer.Amount)
-		if err != nil {
-			return nil, sdkerrors.Wrap(err, "amount")
+func EncodeIBCMsg(portSource types.ICS20TransferPortSource) func(ctx sdk.Context, sender sdk.AccAddress, contractIBCPortID string, msg *wasmvmtypes.IBCMsg) ([]sdk.Msg, error) {
+	return func(ctx sdk.Context, sender sdk.AccAddress, contractIBCPortID string, msg *wasmvmtypes.IBCMsg) ([]sdk.Msg, error) {
+		switch {
+		case msg.CloseChannel != nil:
+			return []sdk.Msg{&channeltypes.MsgChannelCloseInit{
+				PortId:    PortIDForContract(sender),
+				ChannelId: msg.CloseChannel.ChannelID,
+				Signer:    sender.String(),
+			}}, nil
+		case msg.Transfer != nil:
+			amount, err := convertWasmCoinToSdkCoin(msg.Transfer.Amount)
+			if err != nil {
+				return nil, sdkerrors.Wrap(err, "amount")
+			}
+			msg := &ibctransfertypes.MsgTransfer{
+				SourcePort:       portSource.GetPort(ctx),
```

Comment:
:100: and this even picks it up if the port changes after genesis

Reply:
ja, theoretically a parameter change proposal may modify it.